### ISSUE TITLE
Error When Plotting Lines with DateTimes for x values

### ISSUE THREAD

**Issue by xanderdunn** [cphyc/matplotlib-label-lines#15]
matplotlib 3.0.3
matplotlib-label-lines 0.3.6

I am trying to put labels on the lines in the plot where the x axis is datetimes:
![individualImage](https://user-images.githubusercontent.com/1313618/59874857-23487d80-9354-11e9-8152-d0412bcbed7c.png)

And I get this error:
```
    137     set_matplotlib_axes()
    138     figure.autofmt_xdate()
--> 139     labelLines(plt.gca().get_lines(), align=False)
    140 
    141 

/usr/local/lib/python3.5/dist-packages/labellines/core.py in labelLines(lines, align, xvals, **kwargs)
    127 
    128     for line, x, label in zip(labLines, xvals, labels):
--> 129         labelLine(line, x, label, align, **kwargs)

/usr/local/lib/python3.5/dist-packages/labellines/core.py in labelLine(line, x, label, align, **kwargs)
     29     # Find first segment of xdata containing x
     30     for i, (xa, xb) in enumerate(zip(xdata[:-1], xdata[1:])):
---> 31         if min(xa, xb) <= x <= max(xa, xb):
     32             break
     33     else:

TypeError: unorderable types: int() <= datetime.datetime()
  In call to configurable 'train' (<function train at 0x7f8e91b7e400>)
```

**Comment by cphyc** [comment]
Hi,
Thanks for the feedback.

Would it be possible to provide a minimum script that triggers the error?

**Comment by conorroe** [comment]
Hi,
I am trying to plot a graph and label the lines with this code you have developed. is it possible to use this code to add labels to a graph as shown: on the x-axis it is graphed with time intervals in an array. is it possible that i could plot this while including the labels.
![Heating 22°C for 0 5h, 18°C for 1h (Power Curtailment)](https://user-images.githubusercontent.com/61996923/76268048-ab650f00-6264-11ea-93dd-0ce2df79a0c2.png)

**Comment by cphyc** [comment]
Could you please share a minimal working example with some data? 
I never use dates for x axis, so I cannot help you without it.

**Comment by conorroe** [comment]
Thanks for response, I eventually got it to work the way I wanted. Thanks

**Comment by cphyc** [comment]
If so, could you please share it here for future reference?

**Comment by thomasg3** [comment]
I'm runnnig into the same issue but can't get it fixed. Here is the minimal example:

```
example_df = pd.DataFrame({
    'a': np.sin(np.arange(10)) + 1,
    'b': np.arange(10) ** (1/2),
    'c': pd.date_range(start='2020-04-01', periods=10),
})

example_df['a'].plot(label='a')
example_df['b'].plot(label='b')

labelLines(plt.gca().get_lines())

plt.show()

example_df.set_index('c')['a'].plot(label='a')
example_df.set_index('c')['b'].plot(label='b')


labelLines(plt.gca().get_lines()) # This line here fails...

plt.show()
```

The error I get is:
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-187-3edd7105149e> in <module>
      7 _ = plt.legend()
      8 
----> 9 labelLines(plt.gca().get_lines())

.../site-packages/labellines/core.py in labelLines(lines, align, xvals, drop_label, **kwargs)
    152 
    153     for line, x, label in zip(labLines, xvals, labels):
--> 154         labelLine(line, x, label, align, drop_label, **kwargs)

.../site-packages/labellines/core.py in labelLine(line, x, label, align, drop_label, **kwargs)
     42     else:
     43         for i, (xa, xb) in enumerate(zip(xdata[:-1], xdata[1:])):
---> 44             if min(xa, xb) <= x <= max(xa, xb):
     45                 break
     46         else:

pandas/_libs/tslibs/period.pyx in pandas._libs.tslibs.period._Period.__richcmp__()

TypeError: Cannot compare type Period with type datetime
```

**Comment by alanjoshua** [comment]
I am having the same error too. 

`from matplotlib.pyplot import cm
from labellines import labelLine, labelLines
color=iter(cm.rainbow(np.linspace(0,1,len(uniqueStates))))

fig,ax = plt.subplots(figsize=(25,12))
for state in uniqueStates:
    stateData = generalData[generalData['State/UnionTerritory']==state]
    ax.plot(stateData['Date'].to_numpy(),stateData['Confirmed'],label=state,c=next(color))
ax.legend()
ax.tick_params(axis='x', colors='white')
ax.tick_params(axis='y', colors='white')
ax.set_xlabel('Date', color='white',fontsize='x-large',labelpad=20)
ax.set_ylabel('Confirmed cases',color='white',fontsize='x-large',labelpad=20)
ax.set_title('Cummulative confirmed cases of Indian States since January 30',color='white',fontsize='x-large',pad=30)
labelLines(plt.gca().get_lines(), zorder=2.5)
fig.autofmt_xdate()
plt.show()`

The error I am getting is this:

TypeError                                 Traceback (most recent call last)
<ipython-input-248-8bd0e2e81e79> in <module>
     13 ax.set_ylabel('Confirmed cases',color='white',fontsize='x-large',labelpad=20)
     14 ax.set_title('Cummulative confirmed cases of Indian States since January 30',color='white',fontsize='x-large',pad=30)
---> 15 labelLines(plt.gca().get_lines(), zorder=2.5)
     16 fig.autofmt_xdate()
     17 plt.show()

~\Anaconda3\lib\site-packages\labellines\core.py in labelLines(lines, align, xvals, drop_label, **kwargs)
    152 
    153     for line, x, label in zip(labLines, xvals, labels):
--> 154         labelLine(line, x, label, align, drop_label, **kwargs)

~\Anaconda3\lib\site-packages\labellines\core.py in labelLine(line, x, label, align, drop_label, **kwargs)
     42     else:
     43         for i, (xa, xb) in enumerate(zip(xdata[:-1], xdata[1:])):
---> 44             if min(xa, xb) <= x <= max(xa, xb):
     45                 break
     46         else:

TypeError: '<=' not supported between instances of 'int' and 'datetime.datetime'

**Comment by nateGeorge** [comment]
Alright, figured it out with help from this post: https://stackoverflow.com/a/23295415/4549682

The key is using the `set_major_formatter()` command.  You want to plot seconds on the x-axis (as an integer), then do something like the following:

```python
df.plot(legend=False)  # pandas DataFrame with integer seconds since epoch as the index
plt.yscale('log')
labelLines(plt.gca().get_lines())
plt.gca().xaxis.set_major_formatter(
    mtick.FuncFormatter(lambda pos, _: time.strftime("%Y", time.localtime(pos)))
    )
plt.show()
```

In the example, `set_major_locator()` was used, but that wasn't working for me (giving an overflow error) so I guess it's not always needed.

**Comment by cphyc** [comment]
I would be happy to accept any PR that would implement it in the code, unfortunately I will not have time to do it myself in any foreseeable future.

**Comment by nateGeorge** [comment]
The method I showed above doesn't work so great, because you can't set the major tick formatter because matplotlib encounters an overflow error. Instead, all that was needed was to do some checks that the array is a numpy datetime, and then to convert it to a float.

**Comment by cphyc** [comment]
This has been fixed in #29 (thanks @nateGeorge !). Closing for now, but re-open if it doesn't fix. You'll need to update to the latest version 0.3.9.
